rts/Net/AutohostInterface.cpp

```diff
@@ -24,73 +24,131 @@ LOG_REGISTER_SECTION_GLOBAL(LOG_SECTION_AUTOHOST_INTERFACE)
 namespace {
 
 /**
- * @enum EVENT Which events can be sent to the autohost
- *   (in brackets: parameters, where uchar means unsigned char and "string"
- *   means plain ascii text)
+ * @enum EVENT Which events can be sent to the autohost.
+ *
+ * Each packet from engine to autohost interface starts with a byte that
+ * indicated the type of the message. Below in brackets we describe the data
+ * associated with each message.
+ *
+ * char[] are *not* delimited in any way e.g. by '\0', so they can be used only
+ * as a last element of the UDP packet.
  */
 enum EVENT
 {
-	/// Server has started ()
+	/**
+	 * Server has started
+	 *
+	 *   ()
+	 */
 	SERVER_STARTED = 0,
 
-	/// Server is about to exit ()
+	/**
+	 * Server is about to exit
+	 *
+	 *   ()
+	 */
 	SERVER_QUIT = 1,
 
-	/// Game starts ()
+	/**
+	 * Game starts
+	 *
+	 *   (uint32 msgsize, uint8[16] gameId, char[] demoName)
+	 */
 	SERVER_STARTPLAYING = 2,
 
-	/// Game has ended ()
+	/**
+	 * Game has ended
+	 *
+	 *   (uint8 player, uint8 msgsize, uint8[msgsize - 3] winningAllyTeamss)
+	 */
 	SERVER_GAMEOVER = 3,
 
-	/// An information message from server (string message)
+	/**
+	 * An information message from server
+	 *
+	 *   (char[] message)
+	 */
 	SERVER_MESSAGE = 4,
 
-	/// Server gave out a warning (string warningmessage)
+	/**
+	 * A warning message from server
+	 *
+	 *   (char[] warningmessage)
+	 */
 	SERVER_WARNING = 5,
 
-	/// Player has joined the game (uchar playernumber, string name)
+	/**
+	 * Player has joined the game
+	 *
+	 *   (uint8 playernumber, char[] name)
+	 */
 	PLAYER_JOINED = 10,
 
 	/**
-	 * Player has left (uchar playernumber, uchar reason
-	 * (0: lost connection, 1: left, 2: kicked) )
+	 * Player has left
+	 *
+	 *   (uint8 playernumber, uint8 reason)
+	 *
+	 * Reason:
+	 *   - 0: lost connection
+	 *   - 1: left
+	 *   - 2: kicked
 	 */
 	PLAYER_LEFT = 11,
 
 	/**
 	 * Player has updated its ready-state
-	 * (uchar playernumber, uchar state
-	 * (0: not ready, 1: ready, 2: state not changed) )
+	 *
+	 *   (uint8 playernumber, uint8 state)
+	 *
+	 * State:
+	 *   - 0: not ready
+	 *   - 1: ready
+	 *   - 2: forced
+	 *   - 3: failed to ready (in engine code it says it's not clear if possible)
 	 */
 	PLAYER_READY = 12,
 
 	/**
-	 * @brief Player has sent a chat message
-	 *   (uchar playernumber, uchar destination, string text)
+	 * Player has sent a chat message
 	 *
-	 * Destination can be any of: a playernumber [0-32]
-	 * static const int TO_ALLIES = 127;
-	 * static const int TO_SPECTATORS = 126;
-	 * static const int TO_EVERYONE = 125;
-	 * (copied from Game/ChatMessage.h)
+	 *   (uint8 playernumber, uint8 destination, char[] text)
+	 *
+	 * Destination can be any of:
+	 *   - a playernumber
+	 *   - TO_ALLIES = 252
+	 *   - TO_SPECTATORS = 253
+	 *   - TO_EVERYONE = 254
+	 *
+	 * Enum values from rts/Game/ChatMessage.h, value 255 is not special in any
+	 * way, engine swallows all messages to player number >= num active players.
```

Comment:
```suggestion
	 *   - SERVER_PLAYER = 255
```

https://github.com/beyond-all-reason/spring/blob/f295b297d8d39d86c0066ee0670a0e94f86bc16b/rts/Net/GameServer.h#L28

Reply:
Yes, but it's not special in any way in this particular context. Messages send to player 255 end up in the same place as messages send to player 250 (when less then 250 players): nowhere. I've updated the text once again.